### ISSUE TITLE
Wrong error report

### ISSUE THREAD

**Issue by martinduo** [eslint/eslint#14620]
**Tell us about your environment**

Node version: v16.1.0
npm version: v6.14.4
Local ESLint version: Not found
Global ESLint version: vundefined (Currently used)
Operating System: win32 10.0.19042
* **ESLint Version:**
* 7.27.0

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
 "@typescript-eslint/parser": "^4.24.0",

**Please show your full configuration:**
{
  "root": true,
  "env": {
    "node": true,
    "es2021": true
  },
  "extends": ["eslint:recommended"],
  "parserOptions": {
    "ecmaVersion": 12,
    "sourceType": "module"
  },
  "rules": {
    "no-console": 0,
    "no-debugger": 0,
    "prettier/prettier": 0,
    "no-plusplus": 0,
    "no-unused-vars": 0
  }
}

<details>
<summary>Configuration</summary>

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
      let res1 = /(^import.*from )('[.+\/]+\w+')/g.exec(line) // find imported file origin name

```

Fom VSCode:
`  "scripts": {
    "start": "nodemon  src/main.js",
    "lint": "eslint src --color",
     }

```

**What did you expect to happen?**
Expected no lint error. The escape in the "[.+\/]" segment is correct and necessary.

**What actually happened? Please include the actual, raw output from ESLint.**
C:\Users\Bruce\Dropbox\Code\MONEYPLAN\MoneyPlanNodeUtilities\src\main.js
  172:40  error  Unnecessary escape character: \/  no-useless-escape
✖ 1 problem (1 error, 0 warnings)

**Comment by mdjermanovic** [comment]
Hi @martinduo, thanks for the issue!

The `no-useless-escape` rule works as intended in this case. Escaping `/` in character classes (`[]`) is unnecessary.

`/(^import.*from )('[.+/]+\w+')/g` is same as `/(^import.*from )('[.+\/]+\w+')/g`

**Comment by martinduo** [comment]
In my use case, it is necessary. Confirmed by testing here: https://regex101.com/r/cO8lqs/6
I am capturing the name of the exporting file in an import line, e.g.:

import { writeErrorLog, trace, isTraceActive, traceYear } from './moduleTraceFile'

With escape:
![image](https://user-images.githubusercontent.com/879852/119366056-8ea4d380-bc7e-11eb-87d5-302793740836.png)
![image](https://user-images.githubusercontent.com/879852/119367057-93b65280-bc7f-11eb-85a8-2f9ead74e4fb.png)

Without escape:
![image](https://user-images.githubusercontent.com/879852/119366338-daf01380-bc7e-11eb-8783-b5bbd7d5cd0b.png)
![image](https://user-images.githubusercontent.com/879852/119366680-35896f80-bc7f-11eb-980c-999442c27ded.png)

**Comment by mdjermanovic** [comment]
Did you select ECMAScript flavor on regex101.com?

![image](https://user-images.githubusercontent.com/44349756/119370906-22de5d00-bcb6-11eb-9d77-79b851acfabc.png)

**Comment by martinduo** [comment]
No. It was set at PCRE(PHP<7.3. Stupid error by me. I found this page a
year ago and just dived in to build the Regex I needed. Never looked at the
left column. Big mistake. Sorry I bothered you guys. Thanks for the heads
up.
